modules/data-entry/src/main/java/io/uhndata/cards/dataentry/internal/serialize/FlattenFormProcessor.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.uhndata.cards.dataentry.internal.serialize;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import javax.jcr.RepositoryException;
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonValue;
+
+import org.apache.sling.api.resource.Resource;
+import org.osgi.service.component.annotations.Component;
+
+import io.uhndata.cards.serialize.spi.ResourceJsonProcessor;
+
+@Component(immediate = true)
+public class FlattenFormProcessor implements ResourceJsonProcessor
+{
+    // Saves all children with type card:Answer of root node
+    private ThreadLocal<Map<String, JsonObject>> childrenJsons = ThreadLocal.withInitial(HashMap::new);
+    // Saves all children with type card:Answer of specific AnswerSection node
+    private ThreadLocal<JsonObjectBuilder> jsonSection = ThreadLocal.withInitial(Json::createObjectBuilder);
+    // Saves all AnswerSection nodes ids with all children with type card:Answer that belong them
```

Comment:
```suggestion
    // Saves all AnswerSection nodes ids with all children with type cards:Answer that belong them
```
Also, please provide an example of what is being mapped to what. For example, are these AnswerSection node identifiers (JCR Path or `jcr:uuid`?) being mapped to a _list_ or a _hash map_ of child `cards:Answer` nodes?